packages/prettier-plugin-java/src/printers/types-values-and-variables.js

```diff
@@ -1,81 +1,194 @@
 "use strict";
 /* eslint-disable no-unused-vars */
 
+const {
+  concat,
+  join,
+  line,
+  ifBreak,
+  group,
+  indent,
+  dedent
+} = require("prettier").doc.builders;
+const { rejectAndJoin } = require("./printer-utils");
+
 class TypesValuesAndVariablesPrettierVisitor {
   primitiveType(ctx) {
-    return "primitiveType";
+    const annotations = this.mapVisit(ctx.annotation);
+    let type = null;
+    if (ctx.numericType) {
+      type = this.visit(ctx.numericType);
+    } else {
+      type = this.getSingle(ctx).image;
+    }
+
+    return rejectAndJoin(" ", [join(" ", annotations), type]);
   }
 
   numericType(ctx) {
-    return "numericType";
+    return this.visitSingle(ctx);
   }
 
   integralType(ctx) {
-    return "integralType";
+    return this.getSingle(ctx).image;
   }
 
   floatingPointType(ctx) {
-    return "floatingPointType";
+    return this.getSingle(ctx).image;
   }
 
   referenceType(ctx) {
-    return "referenceType";
+    const annotations = this.mapVisit(ctx.annotation);
+
+    const type = ctx.primitiveType
+      ? this.visit(ctx.primitiveType)
+      : this.visit(ctx.classOrInterfaceType);
+
+    const dims = this.visit(ctx.dims);
+
+    return rejectAndJoin("", [join(" ", annotations), " ", type, dims]);
   }
 
   classOrInterfaceType(ctx) {
-    return "classOrInterfaceType";
+    return this.visitSingle(ctx);
   }
 
   classType(ctx) {
-    return "classType";
+    const dots = ctx.Dots;
+
+    if (dots) {
+      const annotations = ctx.annotation;
+      const typeArguments = ctx.typeArguments;
+      const identifiers = ctx.Identifier;
+
+      const segments = [];
+      for (const dot in dots) {
+        const offset = dot.startOffset;
+
+        const currentSegmentAnnotations = [];
+        while (annotations.length > 0) {
+          const annotation = annotations[0];
+          if (annotation && annotation.startOffset < offset) {
+            currentSegmentAnnotations.push(this.visit(annotation));
+            annotations.pop();
+          } else {
+            break;
+          }
+        }
+        const currentSegmentIdentifier = identifiers.pop().image;
+
+        let currentSegmentTypeArgument = undefined;
+        const typeArgument = typeArguments[0];
+        if (typeArgument && typeArgument.startOffset < offset) {
+          currentSegmentTypeArgument = this.visit(typeArgument);
+          typeArguments.pop();
+        }
+
+        segments.push(
+          concat(
+            join(" ", currentSegmentAnnotations),
+            currentSegmentIdentifier,
+            currentSegmentTypeArgument
+          )
+        );
+      }
+
+      return join(".", segments);
+    }
+
+    const annotations = this.mapVisit(ctx.annotation);
```

Comment:
why is the case of less complex name (no dots) not handled as part of the complex case?